src/debug.rs

```diff
@@ -0,0 +1,52 @@
+//! Interacting with debugging agent
```

Comment:
I took those codes from section 5.5.2 of "Debug Target Guide". Normally they are used as default exception handlers, but the main reason I added this call is to control exit codes in qemu.